pytorch_lightning/strategies/ddp.py

```diff
@@ -108,7 +99,7 @@ def __init__(
         self._model_averaging_period = model_averaging_period
         self._pids: Optional[List[int]] = None
         self._sync_dir: Optional[str] = None
-        self._rank_0_has_called_call_children_scripts: bool = False
+        self._rank_0_will_call_children_scripts: bool = False
```

Comment:
Yes, in case if user sets the local rank to the environment, we don't create children scripts and the decision is made inside the launcher. And if it doesn't create the children scripts then we don't need to to set this flag to True.

Another case is when someone calls fit and then test, using the same trainer instance, the second time it launches, we need to keep this as false.